arbor/include/arbor/s_expr.hpp

```diff
@@ -214,6 +214,7 @@ struct ARB_ARBOR_API s_expr {
     s_expr(s_expr l, s_expr r):
         state(pair_type(std::move(l), std::move(r)))
     {}
+    s_expr& operator=(const s_expr& s) { state = s.state; return *this; }
```

Comment:
No, not really. The rule is to implement `op=` if you implement copy and vice veras. The argument is one of correctness:
If you _need_ write one, it'll do something interesting. Therefore, you cannot use the default one.
You'll get a warning, but the result will be wrong, since the assignment operator will still be made for you see
https://gcc.godbolt.org/z/KfreGvxjY
Note especially the difference here
```
S a = S{};     // copy ctor, despite looking like assignment.
S a; a = S{}; // assignment
```